### PR TITLE
git history checker

### LINKED ISSUE #3: Implement git clean commit history checker
The checker has to find git history issues like:

- commit message duplications
- useless commit messages like the following: _fix_, _fixing_, _test_ and others

### PR DESCRIPTION
**Pull request by AndreiPiatrou**
resolves #3

### CODE REVIEW

File: src/git.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const { merge, flow } = require('lodash');
+const gitLog = require('gitlog');
+
+
```
Comment (van1757) [inline src/git.js]: redundant empty line here

File: src/git.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const DEFAULT_OPTIONS = {
+  repo: `${__dirname}/test-repo-folder`,
```
Comment (van1757) [inline src/git.js]: How about using a `path` for this?

File: test/checkers.git.history.artifacts.spec.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+const assert = require('assert');
+
+const buildCheck = require('../src/checkers/git.history.artifacts');
+
+describe('git.history.artifacts.js', () => {
+  it('returns false for not duplicated messages', () => {
```
Comment (van1757) [inline test/checkers.git.history.artifacts.spec.js]: it seems that the title does not correlate with the test result

File: src/checkers/adapters/git.history.adapter.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const { flow } = require('lodash');
+
+function adapt(base, { readGitHistoryFn, onCheckResult }) {
+  return function check(repo, context, config) {
+    return flow(
+      readGitHistoryFn,
+      (history) => {
+        const historyResults = base(history, context, config);
+
+        if (onCheckResult && historyResults.length) {
```
Comment (van1757) [inline src/checkers/adapters/git.history.adapter.js]: Is it necessary to double-check the `historyResults.length` in this and `onCheckResult` functions?

Comment (AndreiPiatrou) [reply src/checkers/adapters/git.history.adapter.js]: for this implementation yes, see related draft PR for more details.

**Comment by AndreiPiatrou** [comment]
@van1757 please, see the updates by your comments. thanks.

**Pull request merged**